wordpress-mobile/WordPress-iOS
# Editor - Move Undo/Redo to the header

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5889
- https://github.com/WordPress/gutenberg/pull/51766
- https://github.com/wordpress-mobile/WordPress-Android/pull/18705

This PR adds new undo/redo buttons in the navigation bar when the editor is opened, it also updates the existing menu icon.

## Screenshots

iPhone Light mode | iPhone Dark mode
-|-
<kbd><img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4885740/4766e0d1-2e13-4e14-bdac-04f02bb1b6aa" width="250" /></kbd> | <kbd><img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4885740/44a5274f-75ea-4060-9004-23d6de359489" width="250" /></kbd>

iPhone Landscape Light mode | iPhone Landscape Dark mode
-|-
<kbd><img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4885740/37db8e0a-c9e8-408a-a966-94b32f6f5e4b" width="250" /></kbd> | <kbd><img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4885740/558a8861-ea44-4ebf-b258-8a36b7afd3c0" width="250" /></kbd>

iPhone RTL |
-|
<kbd><img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4885740/a311549a-2ef1-4091-be32-d6194f331597" width="250" /></kbd> | 

iPad Light mode | iPad Dark mode
-|-
<kbd><img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4885740/9ce33641-4a0d-49f5-aec9-473e1dd9e806" width="250" /></kbd> | <kbd><img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4885740/d9d2e811-af3f-4b01-b964-1ef09591d3a6" width="250" /></kbd>

iPad Landscape Light mode | iPad Landscape Dark mode
-|-
<kbd><img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4885740/dd35f962-ec6f-4971-b6ed-c4526cd03447" width="250" /></kbd> | <kbd><img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4885740/933c1696-5b70-41a3-92a6-53b9eaab4977" width="250" /></kbd>

## To test

- Install the build
- Create a new post
- Add a few blocks/text
- Tap on the new undo/redo buttons in the navigation bar
- **Expect** the content to change accordingly depending on the button you've tapped

**Bonus testing**: [follow these test cases](https://github.com/wordpress-mobile/test-cases/tree/trunk/test-cases/gutenberg/writing-flow#undo--redo---test-cases).

## Regression Notes
1. Potential unintended areas of impact
Post publishing

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Rely on current tests and manual testing

3. What automated tests I added (or what prevented me from doing so)
Added new tests for testing the new undo/redo buttons in the navigation bar.

PR submission checklist:


UI Changes testing checklist:
